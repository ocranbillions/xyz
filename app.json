{
  "addons": [
    "heroku-postgresql"
  ],
  "buildpacks": [
  ],
  "env": {
    "JWT_SECRET_KEY": {
      "required": true
    },
    "NPM_CONFIG_PRODUCTION": {
      "required": true
    }
  },
  "formation": {
  },
  "scripts": {
    "postdeploy": "npx sequelize-cli db:seed:all"
  },
  "name": "xyz",
  "stack": "heroku-18"
}
