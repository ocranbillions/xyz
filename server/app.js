import express from 'express';
import bodyParser from 'body-parser';
import { config } from 'dotenv';
import userRoutes from './routes/userRoutes';

config();

const app = express();

const port = process.env.PORT || 3000;

app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));

// app.get('/', (req, res) => res.status(301).redirect('/api/v1'));

app.get('/', (req, res) => {
  res.send('Welcome to xzy');
});

// Handles user routes
app.use('/api/v1/users', userRoutes);


// app.use(ErrorHandler.sendError);


if (process.env.NODE_ENV !== 'test') {
  app.listen(port, () => console.log('Welcome to xyz'));
}
export default app;
