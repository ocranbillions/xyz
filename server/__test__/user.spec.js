/* eslint-disable no-undef */
import supertest from 'supertest';
// import sinon from 'sinon';
import app from '../app';

// import models from '../db/models';
// import userCtrl from '../controllers/userController'

const server = () => supertest(app);
const url = '/api/v1';

describe('Signup endpoint', () => {
  it('Should register new user', (done) => {
    server().post(`${url}/auth/signup`).send({
      firstName: "sam",
      lastName: "ocr",
      email: "samm@gmail.com",
      password: 'password123..',
    })
      .end((err, res) => {
        expect(res.body.status).toEqual(201);
        expect(res.body.data).toHaveProperty('token');
        expect(res.body.data).toHaveProperty('id');
        expect(res.body.data).toHaveProperty('firstName');
        expect(res.body.data).toHaveProperty('lastName');
        expect(res.body.data).toHaveProperty('email');
        done();
        expect(res.body).toMatchSnapshot();
      });
  });

  // it('should fail', (done) => {
  //   sinon.stub(models.Users, 'findOne');
  //   models.Users.findOne.throws();

  //   const req = {
  //     body: {
  //       firstName: 'jsfgfg',
  //       lastName: 'dfdfdf',
  //       email: 'test@test.com',
  //       password: 'tester',
  //     },
  //   };

  //   // const res = {
  //   //   status: (num) => {
  //   //     return num;
  //   //   },
  //   //   // status: 500,
  //   //   body: {
  //   //     firstName: 'jsfgfg',
  //   //     lastName: 'dfdfdf',
  //   //     email: 'test@test.com',
  //   //     password: 'tester',
  //   //   },
  //   // };

  //   const res = {
  //     statusCode: 500,
  //     userStatus: null,
  //     status: function(code) {
  //       this.statusCode = code;
  //       return this;
  //     },
  //     json: function(data) {
  //       this.userStatus = data.status;
  //     }
  //   };

  //   userCtrl.signUp(req, res, () => {}).then((response) => {
  //     console.log(response);
  //     expect(response.statusCode).toEqual(500);
  //     // expect(result).to.be.an('error');
  //     // expect(result).to.have.property('statusCode', 500);
  //     done();
  //   });

  //   models.Users.findOne.restore();
  // });
});
