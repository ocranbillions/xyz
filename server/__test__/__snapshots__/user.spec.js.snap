// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup endpoint Should register new user 1`] = `
Object {
  "data": Object {
    "email": "samm@gmail.com",
    "firstName": "sam",
    "id": 5,
    "lastName": "ocr",
    "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NSwiZmlyc3ROYW1lIjoic2FtIiwibGFzdE5hbWUiOiJvY3IiLCJlbWFpbCI6InNhbW1AZ21haWwuY29tIiwiaWF0IjoxNTYyOTM4NzA0LCJleHAiOjE1NjMwMjUxMDR9.6VzOf25MJ_HDtOnDl9sPd4kvgA0YXQSaimmcKLPU25M",
  },
  "message": "User Created successfully",
  "status": 201,
}
`;
